src/components/Form/InputWrapper.tsx

```diff
@@ -1,21 +1,66 @@
 import type {ForwardedRef} from 'react';
 import React, {forwardRef, useContext} from 'react';
 import type {AnimatedTextInputRef} from '@components/RNTextInput';
+import RoomNameInput from '@components/RoomNameInput';
 import TextInput from '@components/TextInput';
+import {canUseTouchScreen} from '@libs/DeviceCapabilities';
 import FormContext from './FormContext';
 import type {InputWrapperProps, ValidInputs} from './types';
 
-function InputWrapper<TInput extends ValidInputs>({InputComponent, inputID, valueType = 'string', ...rest}: InputWrapperProps<TInput>, ref: ForwardedRef<AnimatedTextInputRef>) {
+function computeComponentSpecificRegistrationParams<TInput extends ValidInputs>({
+    InputComponent,
+    shouldSubmitForm,
+    multiline,
+    autoGrowHeight,
+    blurOnSubmit,
+}: InputWrapperProps<TInput>): {
+    readonly shouldSubmitForm: boolean;
+    readonly blurOnSubmit: boolean | undefined;
+    readonly shouldSetTouchedOnBlurOnly: boolean;
+} {
+    const textInputBasedComponents = [TextInput, RoomNameInput] as TInput[];
```

Comment:
Type assertions should be avoided at a high cost. They are a very dangerous type system feature.

As I understand this, this typing will also work:

```ts
const textInputBasedComponents: ValidInputs[] = [TextInput, RoomNameInput];
```

Reply:
Thanks for the suggestion. I was unable to use this correctly at first.